packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -534,27 +546,40 @@ export function TransferPanel() {
   }
 
   const isTransferAllowed = useMemo(() => {
+    const isConnectedToTheWrongChain =
+      latestChain.current?.chain?.id !== latestNetworks.current.sourceChain.id
+
+    if (!arbTokenBridgeLoaded) {
+      return false
+    }
+    if (!latestEth.current) {
+      return false
+    }
     if (!isConnected) {
       return false
     }
     if (!walletAddress) {
       return false
     }
+    if (isConnectedToTheWrongChain) {
+      return false
+    }
     if (!!destinationAddressError) {
       return false
     }
     return true
-  }, [destinationAddressError, isConnected, walletAddress])
+  }, [
+    arbTokenBridgeLoaded,
+    destinationAddressError,
+    isConnected,
+    latestChain,
+    latestEth,
+    latestNetworks,
+    walletAddress
+  ])
 
   const transfer = async () => {
-    try {
-      setTransferring(true)
-      if (chainId !== networks.sourceChain.id) {
-        await switchNetworkAsync?.(networks.sourceChain.id)
-      }
-    } finally {
-      setTransferring(false)
-    }
+    const sourceChainId = latestNetworks.current.sourceChain.id
 
     if (!isTransferAllowed) {
```

Comment:
when user reached here, the connected chain must be the source chain, or else it won't proceed.